winlogbeat/eventlog/wineventlog.go

```diff
@@ -159,12 +160,12 @@ func (l *winEventLog) IsFile() bool {
 	return l.file
 }
 
-func (l *winEventLog) Open(state checkpoint.EventLogState) error {
+func (l *winEventLog) Open(state checkpoint.EventLogState, metricsRegistry *monitoring.Registry) error {
 	l.lastRead = state
 	// we need to defer metrics initialization since when the event log
 	// is used from winlog input it would register it twice due to CheckConfig calls
-	if l.metrics == nil {
-		l.metrics = newInputMetrics(l.channelName, l.id)
+	if l.metrics == nil && l.id != "" {
```

Comment:
is the `l.id` check still necessary?  

in `newWinEventLog` we do:

```
	id := c.ID
	if id == "" {
		id = c.Name
	}
```

So we shouldn't have a zero value string for id, but if the config didn't have ID we would use the channel name.

Reply:
I think you're right, but for a one-day-turnaround urgent bugfix release my inclination is to keep tangential logic 1-1 even if it seems redundant, and `newInputMetrics` previously gated on that check